### DISCUSSION TITLE
Evaluate raw 1074 continuum images to quantify spillage and guide occulter correction

### DISCUSSION THREAD

**Discussion by detoma** [NCAR/ucomp-pipeline#291] [General]
UCOMP images are over-occulted by 10-14 pixels depending on the time of the year.

A serious problem in the UCOMP data is the Sun is often not well centered under the occulter. 

According to Ben when a manual occulter adjustment is performed by the observers, the accuracy is +/-2pixels. These adjustments are done typically only once a day. It is not uncommon for the Sun to be off 6-7 pixels before an adjustment is made. 

Wind and seeing can cause image motion on very short time scales that we cannot correct for. There is also a slow drift, mostly in the detector x-direction, over the day. This slow drift is something we can evaluate by examining the bright ring around the raw continuum images and try to correct for.

Ben identified two images that were taken before and after an alignment shown below. It is clear visually that the "before alignment" image has an asymmetric bright ring and more bright pixels than the "after alignment" image.

raw image before alignment at 20221016  22:51:00 (no dark removed, no hot pixel correction)
<img width="1279" alt="Screenshot 2024-07-24 at 5 28 20 PM" src="https://github.com/user-attachments/assets/21e31f27-40cb-4390-8141-d4cc7eb492d3">

raw image after alignment at 20221016  22:54:36 (no dark removed, no hot pixel correction)
<img width="1279" alt="Screenshot 2024-07-24 at 5 28 37 PM" src="https://github.com/user-attachments/assets/dcb14ea5-8812-4bec-a561-25c2f75597e9">

When a threshold of 2500DN, which is in the non-linear regime and close to saturation, is set we find that the "before alignment image" has 3922 above 250DN and the "one after the alignment has 2777 pixels, which confirm what is seen visually. Pixels with raw values above ~2500 cannot be properly calibrated.

The difference is best visible in the very bright pixels. Below is an histogram od the bright pixels for the image before alignment in white and after alignment in red.
<img width="1185" alt="Screenshot 2024-07-24 at 5 02 56 PM" src="https://github.com/user-attachments/assets/ffcc92ab-9769-4158-99ce-f5eaab335e46">

**Comment by detoma** [comment]
Below are mask of the images above only showing the pixels above 2500 DN. Note that some hot pixels show up in these masks. Also note that the Sun is not perfectly centered under the occulter even after the alignment is performed. The image "before alignment" has a clear asymmetry left/right,  the one "after alignment" has a smaller asymmetry top/bottom.

"before alignment"
<img width="1272" alt="Screenshot 2024-07-25 at 12 28 56 AM" src="https://github.com/user-attachments/assets/de4e4f42-fcc8-4c24-92f5-fb2e8b8bca54">

"after alignment"
<img width="1275" alt="Screenshot 2024-07-25 at 12 29 10 AM" src="https://github.com/user-attachments/assets/1658c39c-93c3-4b79-8e51-00c0d6968ac3">

The number of pixels > 2500DN on the right and left side of the image "before alignment" are: 357 and 3130. The ratio is 8
The number of pixels > 2500DN on the right and left side of the image "after alignment" are: 963 and 1275. The ratio is 1.32
The number of pixels >2500DN in the top and bottom part of the image "before alignment" are: 2769 and 715. The ratio is 3.9
The number of pixels >2500DN in the top and bottom part of the image "after alignment" are: 647 and 1589. The ratio is 0.41

The right, left, top, and bottom above are defined relative to the center of the occulter, not the center of the detector. This step is necessary to properly evaluate the unbalance around the occulter but requires finding the center of occulter.

**Comment by bberkeyU** [comment]
Is the second image well aligned? There are more bright pixels in the lower left than in the upper right.  

The top image has ~40% more 2500DN> pixels than the bottom, which sounds like we are now thinking of it as misaligned. Assuming the bottom is well aligned, how far can we get from this before it is also considered misaligned?  

GdT: Neither of the two images is well aligned. The one "before alignment" is worse.  The image before alignment has 7 times more pixels in the left side than the right. This is definitely misaligned.

**Comment by detoma** [comment]
We need to discuss what is a tolerable offset and how to apply correction.  

We cannot correct for seeing because it too variable. What we can do in real time is to compensate for slow motions. Each image has seeing in it. If we try to correct for the shifts we see in each image, we will transfer seeing from one image to the next and will make images jump all over.
